registry/remote/utils.go

```diff
@@ -71,3 +71,14 @@ func withScopeHint(ctx context.Context, ref registry.Reference, actions ...strin
 	scope := auth.ScopeRepository(ref.Repository, actions...)
 	return auth.AppendScopes(ctx, scope)
 }
+
+// Helper function for TestRegistry_Repositories_WithLastParam
+// and TestRepository_Tags_WithLastParam
+func indexOf(element string, data []string) int {
+	for ind, val := range data {
+		if element == val {
+			return ind
+		}
+	}
+	return -1 //not found.
+}
```

Comment:
Since this function is for testing purpose, try to put it in `_test.go` files or even create `utils_test.go` but never put it in regular go files.

Reply:
> Since this function is for testing purpose, try to put it in `_test.go` files or even create `utils_test.go` but never put it in regular go files.

utils_test.go sounds like a place to put test functions for the utils.go file. I'll move the helper function back to its original _test.go file.